contracts/JBController.sol

```diff
@@ -708,12 +702,12 @@ contract JBController is IJBController, JBOperatable, ReentrancyGuard {
     if (uint256(_processedTokenTrackerOf[_projectId]) != tokenStore.totalSupplyOf(_projectId))
       _distributeReservedTokensOf(_projectId, '');
 
-    // Make sure the new controller is prepped for the migration.
-    _to.prepForMigrationOf(_projectId, this);
-
     // Set the new controller.
     directory.setControllerOf(_projectId, _to);
 
+    // Make sure the new controller is prepped for the migration.
+    _to.prepForMigrationOf(_projectId, this);
```

Comment:
prepForMigration should be called prior to the terminal being prepped being the active controller. 

See line 71. This check is important to make sure any resetting logic doesn't get called while a project has an in-progress tokenTracker.